src/FastRouteRouter.php

```diff
@@ -256,10 +261,17 @@ private function marshalMatchedRoute(array $result, $method)
             return RouteResult::fromRouteFailure();
         }
 
+        $params = $result[2];
+
+        $options = $route->getOptions();
+        if (! empty($options['defaults'])) {
+            $params = array_merge($options['defaults'], $params);
```

Comment:
For consistency with the other implementations, I recommend using the `DefaultParamsTrait` from zend-expressive-template; that trait also supplies a recursive merge routine that's more robust.